src/NServiceBus.Core/Outbox/OutboxDeduplicationBehavior.cs

```diff
@@ -63,23 +58,35 @@ void DispatchOperationToTransport(IEnumerable<TransportOperation> operations)
 
                 deliveryOptions.EnlistInReceiveTransaction = false;
 
-                var message = new OutgoingMessage(transportOperation.MessageId,transportOperation.Headers, transportOperation.Body);
+                var message = new OutgoingMessage(transportOperation.MessageId, transportOperation.Headers, transportOperation.Body);
 
-                //dispatch to transport
+                var operationType = transportOperation.Options["Operation"];
```

Comment:
I think a TransportOperation abstraction might be suitable, Publish/Send/Defer.

We should also wait for @johnsimons to complete the removal of bus.Return since that will clean up the code massively and that might uncover a better abstraction as well